From cafbd631622258741d4c42caaf9a52b8fd4db626 Mon Sep 17 00:00:00 2001
From: Jean-Marie Lemetayer <jean-mariex.lemetayer@intel.com>
Date: Tue, 3 Jul 2012 16:02:24 +0200
Subject: [PATCH 16/78] WLAN: Immediate authentication on beacon IE

BZ: 42986

The fix in the driver changes the NLCP behavior so it's now similar
to MCP. Now we go with immediate authentication if either probe_rsp_ies
OR beacon_ies exists in the supplicant's request.

Change-Id: Ia739c23a2be596e95b6f9a96e07eaf8c517dfc25
Orig-Change-Id: Ic9d49451359346307c0a8a0c12194e35a2b818bf
Signed-off-by: Jean-Marie Lemetayer <jean-mariex.lemetayer@intel.com>
Reviewed-on: http://android.intel.com:8080/61812
Reviewed-by: Regairaz, Julie <julie.regairaz@intel.com>
Reviewed-by: De Cesco, Jonathan <jonathan.de.cesco@intel.com>
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: Yin, Fengwei <fengwei.yin@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
Reviewed-on: http://android.intel.com:8080/77973
Reviewed-by: Li, XiaojingX <xiaojingx.li@intel.com>
Tested-by: Li, XiaojingX <xiaojingx.li@intel.com>
---
 wl12xx-compat/net/mac80211/mlme.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/wl12xx-compat/net/mac80211/mlme.c b/wl12xx-compat/net/mac80211/mlme.c
index 08d6538..962f072 100644
--- a/wl12xx-compat/net/mac80211/mlme.c
+++ b/wl12xx-compat/net/mac80211/mlme.c
@@ -2525,7 +2525,7 @@ int ieee80211_mgd_auth(struct ieee80211_sub_if_data *sdata,
 	wk->probe_auth.privacy = req->bss->capability & WLAN_CAPABILITY_PRIVACY;
 
 	/* if we already have a probe, don't probe again */
-	if (req->bss->proberesp_ies)
+	if (req->bss->proberesp_ies || req->bss->beacon_ies)
 		wk->type = IEEE80211_WORK_AUTH;
 	else
 		wk->type = IEEE80211_WORK_DIRECT_PROBE;
-- 
1.7.9

